### PR TITLE
Adds support for the AudioInjector.net Octo sound card.

### PR DESCRIPTION
**Pull request by flatmax**
Contains the sound/soc/bcm ALSA machine driver and necessary alterations
to the Kconfig and Makefile. Adds the dts overlay and updates the Makefile
and README.
Updates the relevant defconfig files to enable building for the Raspberry Pi.

### CODE REVIEW

File: sound/soc/bcm/audioinjector-octo-soundcard.c [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+#include <sound/soc.h>
+#include <sound/pcm_params.h>
+#include <sound/control.h>
+
+static struct gpio_descs *mult_gpios;
+static unsigned int audioinjector_octo_rate;
+
+static int audioinjector_octo_soundcard_dai_init(struct snd_soc_pcm_runtime *rtd)
+{
+  return snd_soc_dai_set_bclk_ratio(rtd->cpu_dai, 64);
```
Comment (pelwell) [inline sound/soc/bcm/audioinjector-octo-soundcard.c:34]: Stick to the Linux indentation rules - hard TAB characters, set to 8 characters.

File: sound/soc/bcm/audioinjector-octo-soundcard.c [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+	if ((ret = snd_soc_register_card(card)))
+		dev_err(&pdev->dev, "snd_soc_register_card failed (%d)\n", ret);
+	return ret;
+}
+
+static int audioinjector_octo_soundcard_remove(struct platform_device *pdev)
+{
+	struct snd_soc_card *card = platform_get_drvdata(pdev);
+	return snd_soc_unregister_card(card);
+
```
Comment (pelwell) [inline sound/soc/bcm/audioinjector-octo-soundcard.c:198]: Delete this trailing blank line.

File: sound/soc/bcm/audioinjector-octo-soundcard.c [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+       .probe          = audioinjector_octo_soundcard_probe,
+       .remove         = audioinjector_octo_soundcard_remove,
+};
+
+module_platform_driver(audioinjector_octo_soundcard_driver);
+MODULE_AUTHOR("Matt Flax <flatmax@flatmax.org>");
+MODULE_DESCRIPTION("AudioInjector.net octo Soundcard");
+MODULE_LICENSE("GPL v2");
+MODULE_ALIAS("platform:audioinjector-octo-soundcard");
+
```
Comment (pelwell) [inline sound/soc/bcm/audioinjector-octo-soundcard.c:222]: Delete this trailing blank line.

File: arch/arm/boot/dts/overlays/Makefile [hunk 4]
```diff
@@ -19,6 +19,7 @@ dtbo-$(RPI_DT_OVERLAYS) += akkordion-iqdacplus.dtbo
 dtbo-$(RPI_DT_OVERLAYS) += allo-piano-dac-pcm512x-audio.dtbo
 dtbo-$(RPI_DT_OVERLAYS) += at86rf233.dtbo
 dtbo-$(RPI_DT_OVERLAYS) += audioinjector-wm8731-audio.dtbo
+dtbo-$(RPI_DT_OVERLAYS) += audioinjector-addons.dtbo
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/Makefile:22]: I know it's picky, but can you put this in front of the previous line (in lexicographic order).

File: arch/arm/boot/dts/overlays/README [hunk 5]
```diff
@@ -273,6 +273,12 @@ Load:   dtoverlay=audioinjector-wm8731-audio
 Params: <None>
 
 
+Name:   audioinjector-addons
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/README:276]: Order, again.

File: arch/arm/boot/dts/overlays/audioinjector-addons-overlay.dts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+				clock-frequency = <49152000>;
+			};
+		};
+	};
+
+	fragment@2 {
+		target = <&sound>;
+		__overlay__ {
+			compatible = "ai,audioinjector-octo-soundcard";
+      mult-gpios = <&gpio 27 0>, <&gpio 22 0>, <&gpio 23 0>;
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/audioinjector-addons-overlay.dts:43]: Indentation.

**Review by pelwell — CHANGES_REQUESTED** [review]
I've made a few largely nit-picky comments. The biggest issue is that this PR is against 4.4 and we've now moved to 4.9 (sorry about the review delay).

If you fix the nits I can merge against 4.4 and then cherry-pick onto 4.9, or you can go straight to 4.9.

**Comment by flatmax** [comment]
Hey Phil,

I think it will be easiest if I kill this pull request. I will switch to 
the 4.9 branch and run check_patch.pl on my changes before submitting - 
apologies for not doing that in the first place.

Matt

**Comment by pelwell** [comment]
Resubmitted and merged as #1883.

**Pull request closed without merging**